reactive/k8s.py

```diff
@@ -11,10 +13,25 @@
 from charmhelpers.core.hookenv import status_set
 from charmhelpers.core.templating import render
 
-from os import getenv
 from contextlib import contextmanager
 
 
+@hook('config-changed')
+def config_changed():
+    '''If the configuration values change, remove the available state.'''
+    config = hookenv.config()
+    if any(config.changed(key) for key in config.keys()):
+        hookenv.log('Configuration options have changed.')
```

Comment:
We consider anytime you're using `hookenv.log` to instead use `hookenv.status_set`, `log` is way less visible to the user and should be there to just to add supplementary debug data. having a `hookenv.log('Configuration options have changed')` and then a `hookenv.status_set('maintenance', 'configuring kubernetes')` so the operator sees what's going on while using log to really highlight the lower level sutff.

To make a really crap analogy to project management. `status-set` is like an story, then `log` are all the small tasks that build that story. We should try to group and model that as we build out layers